lib/msf/core/exploit/remote/smb/relay/ntlm/server.rb

```diff
@@ -0,0 +1,62 @@
+module Msf::Exploit::Remote::SMB::Relay::NTLM
+  # This class provides the SMB server core. Settings that are relevant server wide are managed by this object.
+  # Currently, the server only supports negotiating and authenticating requests. No other server functionality is
+  # available at this time. The negotiating and authentication is supported for SMB versions 1 through 3.1.1.
+  class Server < ::RubySMB::Server
+    def initialize(relay_timeout:, relay_targets:, listener:, thread_manager:, **kwargs)
+      super(**kwargs)
+
+      @relay_targets = relay_targets
```

Comment:
So from my testing and what I remember from our conversations, SMB 3.1.1 can't be relayed right? When I try to connect from Server 2019 and RubySMB which both attempt to negotiate 3.1.1, the messages just say "New request from", log some details then nothing.

Adding this line here to remove the 3.1.1 dialect from the server seems to be sufficient to ensure that it's negotiating 3.0.2 at the latest and can thus relay the incoming connections in those scenarios. If we truly can't relay 3.1.1 because of the new pre-auth capabilities in it, then perhaps we should consider disabling it on the server.

```suggestion
      @dialects.delete('0x0311')
      @relay_targets = relay_targets
```

Reply:
Good catch, thanks! 🎉 

The relay client from B to C was updated to not request 0x0311 -  but not the server from A to B 😄